File: Backend/models/usuario.py
from sqlmodel import Field, SQLModel, Relationship
from typing import Optional, List
import datetime
from pydantic import BaseModel


class UsuarioBase(SQLModel):
    Nombre: str
    Apellido: str
    Correo: str = Field(unique=True, index=True)  

class Usuario(UsuarioBase, table=True):
    IdUsuario: Optional[int] = Field(default=None, primary_key=True)
    Contrasena: str
    FechaRegistro: datetime.date = Field(default_factory=datetime.datetime.utcnow)
    estado: bool = True

    busfavorito:Optional["BusFavorito"]= Relationship(back_populates="usuario")

class UsuarioCreate(UsuarioBase):
    Contrasena: str

class UsuarioPublic(UsuarioBase):
    IdUsuario: int
    FechaRegistro: datetime.date
    estado: bool

class UsuarioUpdate(SQLModel):
    Nombre: Optional[str] = None
    Apellido: Optional[str] = None
    Correo: Optional[str] = None
    Contrasena: Optional[str] = None
    estado: Optional[bool] = None

File: flutter/lib/data/models/usuario.dart
class Usuario {
  final int idUsuario;
  final String nombre;
  final String apellido;
  final String correo;
  final String contrasena;
  final DateTime fechaRegistro;
  final bool estado;

  const Usuario({
    required this.idUsuario,
    required this.nombre,
    required this.apellido,
    required this.correo,
    required this.contrasena,
    required this.fechaRegistro,
    required this.estado,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      idUsuario: json['IdUsuario'] ?? 0,
      nombre: json['Nombre'] ?? "",
      apellido: json['Apellido'] ?? "",
      correo: json['Correo'] ?? "",
      contrasena: json['Contrasena'] ?? "",
      fechaRegistro: DateTime.parse(json['FechaRegistro']),
      estado: json['Estado'] ?? true,
    );
  }

  Map<String, dynamic> toJson() => {
    'IdUsuario': idUsuario,
    'Nombre': nombre,
    'Apellido': apellido,
    'Correo': correo,
    'Contrasena': contrasena,
    'FechaRegistro': fechaRegistro.toIso8601String(),
    'Estado': estado,
  };
}
